UnleashClient/streaming/connector.py

```diff
@@ -0,0 +1,139 @@
+import threading
+import time
+from typing import Callable, Optional
+
+from ld_eventsource import SSEClient
+from ld_eventsource.config import ConnectStrategy, ErrorStrategy, RetryDelayStrategy
+
+from UnleashClient.constants import STREAMING_URL
+from UnleashClient.utils import LOGGER
+
+from .event_processor import StreamingEventProcessor
+
+
+class StreamingConnector:
+    """
+    Manages the SSE connection lifecycle with reconnect/backoff and delegates
+    event handling to an injected StreamingEventProcessor.
+    """
+
+    def __init__(
+        self,
+        url: str,
+        headers: dict,
+        request_timeout: int,
+        event_processor: StreamingEventProcessor,
+        on_ready: Optional[Callable[[], None]] = None,
+        sse_client_factory: Optional[Callable[[str, dict, int], SSEClient]] = None,
+        heartbeat_timeout: int = 60,
+        backoff_initial: float = 2.0,
+        backoff_max: float = 30.0,
+        backoff_multiplier: float = 2.0,
+        backoff_jitter: Optional[float] = 0.5,
+        custom_options: Optional[dict] = None,
+    ) -> None:
+        self._base_url = url.rstrip("/") + STREAMING_URL
+        self._headers = {**headers, "Accept": "text/event-stream"}
+        self._timeout = request_timeout
+        self._on_ready = on_ready
+        self._sse_factory = sse_client_factory
```

Comment:
creating testability/swapping option is only valuable if you gonnd use this option. From my review it looks like we exposed this option (increased complexity) but we never exercised this option. I think in the production implementation of a connector.py we only need to wrap LD client. More detail why too much dependency inversion can be harmful: https://dannorth.net/blog/cupid-the-back-story/#dependency-inversion-principle

Reply:
+1 for this. Too much injection is how you end up with a need for IoC containers *shudder*. No one wants that